python/ray/serve/_private/application_state.py

```diff
@@ -954,8 +957,16 @@ def override_deployment_info(
         return deployment_infos
 
     config_dict = override_config.dict(exclude_unset=True)
+    app_logging_config = config_dict.get("logging_config", None)
     deployment_override_options = config_dict.get("deployments", [])
 
+    # If application loggin config is set, apply to all existing deployment config
+    if app_logging_config:
+        for _, info in deployment_infos.items():
+            info.deployment_config.logging_config = LoggingConfig(
+                **app_logging_config
+            ).dict()
```

Comment:
Won't this override all deployments' logging config, even if there is one specified for the deployment? The desired behavior is being able to override deployment > application > system (this is standard practice, finest granularity should take precedence).

Reply:
Hmm from the tests it looks like what I'm saying is already the case. I'm not following how this works because it looks like you're overriding all deployments' logging config here with the application one.

Reply:
Apply the application logging config here.
And then we have the for loop to override the deployment config if set. https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/application_state.py#L1020

So that deployment logging config will always be set lastly.

Reply:
That was my confusion as well

Reply:
synced with Cindy offline, we apply the application config out of the function. So that we don't need to track whether setting in the deployment code or yaml file.

Reply:
Got it. So this works as-is? Can you leave a comment describing this btw it's pretty hard to see from the code

Reply:
I need to write a unit test to cover this case. Yeah, I will leave comments too!